shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -244,6 +233,25 @@ def _scan_product__select_move_line(
                 stock.mark_move_line_as_picked(move_line)
             return self._response_for_set_quantity(move_line)
 
+    def _select_move_line_from_product(self, product, location, package, lot):
+        domain = self._scan_product__select_move_line_domain(
+            product, location=location, package=package, lot=lot
+        )
+        query = self.env["stock.move.line"]._search(domain, limit=1)
```

Comment:
as order by is overwritten afterwards, don't apply default order by
```suggestion
        query = self.env["stock.move.line"]._search(domain, order="id", limit=1)
```
Default order by is on "result_package_id" which is a non required field so it will cause a useless nullable join if you don't remove it from the search. Nullable joins are not allowed in FOR UPDATE